llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -1351,6 +1351,27 @@ class VPInstruction : public VPRecipeWithIRFlags,
     }
   }
 
+  /// Returns true if the underlying opcode may read from or write to memory.
+  bool opcodeMayReadOrWriteFromMemory() const {
+    if (Instruction::isBinaryOp(getOpcode()))
```

Comment:
Would probably be better to move this to VPlanRecipes.cpp, so it is slightly easier to look up from the uses in the file

And also make private, to encourage the other general helpers to be used.

Reply:
I think I tried making it private/protected earlier, but then it means we can't access it from `VPRecipeBase::mayWriteToMemory`/`VPRecipeBase::mayReadToMemory`. Unless we override those instead in VPInstruction?